errors.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package goacme
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+// ErrorType is used to define a set of errors predefined by the ACME spec.
+type ErrorType string
+
+const (
+	ErrBadCSR       = "urn:acme:error:badCSR"         // The CSR is unacceptable (e.g., due to a short key)
```

Comment:
Precisely: it's not possible to define global "const" errors and have them pass variable details.

The reason is when you do e.g. `if err == io.EOF { ... }` it actually compares struct pointers, just like in C, only implicit. So, if one wanted to have a pre-defined error with variable message, the error would have to be copied (to a new memory address) which sort of defeats the purpose of it all since `err` and `MyError` in `if err == MyError` will never match: err will now have a different memory address.

But, you got me thinking: the errors predefined by ACME spec seem quite stable. For instance, `urn:acme:error:badNonce` is "Unacceptable anti-replay nonce", there's not much else to say in "details". In other words, handling of such pre-defined errors will unlikely depend on the details field.

So, I went on and actually declared pre-defined errors as structs, just to see how it would look like.